packages/shared/src/components/modals/streaks/NewStreakModal.tsx

```diff
@@ -0,0 +1,129 @@
+import { Checkbox } from '../../fields/Checkbox';
+import { useActions } from '../../../hooks';
+import { ActionType } from '../../../graphql/actions';
+import { ModalClose } from '../common/ModalClose';
+import { cloudinary } from '../../../lib/image';
+
+interface FirstStreakModalProps extends LazyModalCommonProps {
+  currentStreak: number;
+  previousStreak: number;
+}
+
+const Paragraph = classed('p', 'text-center text-theme-label-tertiary');
+
+export default function FirstStreakModal({
+  currentStreak,
+  previousStreak,
+  onRequestClose,
+  ...props
+}: FirstStreakModalProps): ReactElement {
+  const [isChecked, setIsChecked] = useState(false);
+  const { completeAction } = useActions();
+  const shouldShowSplash = currentStreak > 20;
+  const onClose = (e: React.MouseEvent | React.KeyboardEvent) => {
+    if (isChecked) {
+      completeAction(ActionType.OptOutStreaks);
+    }
+
+    onRequestClose(e);
+  };
+
+  return (
+    <Modal
+      {...props}
+      kind={Modal.Kind.FlexibleTop}
+      size={Modal.Size.XSmall}
+      onRequestClose={onClose}
+    >
+      <ModalClose onClick={onRequestClose} className="right-2 top-2" />
+      <Modal.Body className="items-center">
+        <span className="relative flex flex-col items-center justify-center">
+          {shouldShowSplash ? (
+            <img
+              src={cloudinary.streak.splash}
+              alt="A splash design for background"
+              className="ml-2 h-[10rem] w-[15rem] text-theme-color-bacon"
+            />
+          ) : (
+            <FireIcon className="h-[10rem] w-[10rem] text-theme-color-bacon" />
+          )}
+          <strong className="typo-tera absolute">{currentStreak}</strong>
+          {shouldShowSplash && (
+            <span className="typo-tera absolute mt-44">🏆</span>
+          )}
+        </span>
+        <strong
+          className={classNames(
+            'typo-title1',
+            shouldShowSplash ? 'mt-20' : 'mt-10',
+          )}
+        >
+          {shouldShowSplash
+            ? 'New streak record!'
+            : `${currentStreak} days streak`}
+        </strong>
+        <Paragraph
+          className={classNames(
+            'typo-body',
+            shouldShowSplash ? 'mt-3' : 'mt-5',
+          )}
+        >
+          {shouldShowSplash
+            ? 'Epic win! You are on a league of your own'
+            : `New milestone reached! You are unstoppable. Your previous record was
+        ${previousStreak} days.`}
+        </Paragraph>
+        <Paragraph className="mt-10 typo-callout">
+          <strong>Brag about it via...</strong>
+        </Paragraph>
+        <div className="mt-3 flex flex-row gap-2">
+          <SocialShareButton
+            href="" // what should be the link
+            icon={<TwitterIcon />}
+            variant={ButtonVariant.Primary}
+            color={ButtonColor.Twitter}
+            label="X"
+          />
+          <SocialShareButton
+            href="" // what should be the link
```

Comment:
I think there was a comment on this from Chris, but can't see it while reviewing. So just making sure we'll have this as a todo with a ticket NR if it remains unresolved for this PR merge.